src/mjolnir/graphvalidator.cc

```diff
@@ -557,7 +553,17 @@ void GraphValidator::Validate(const boost::property_tree::ptree& pt) {
   std::shuffle(tilequeue.begin(), tilequeue.end(), std::mt19937(3));
 
   // Remember what the dataset id is in case we have to make some tiles
-  auto dataset_id = GraphTile(tile_dir, *tilequeue.begin()).header()->dataset_id();
+
+  uint64_t dataset_id = 0;
+  if (const auto& tile = GraphTile::Create(tile_dir, *tilequeue.begin())) {
+    if (const auto* header = tile->header()) {
+      dataset_id = header->dataset_id();
+    }
+  }
+  if (dataset_id == 0) {
+    LOG_ERROR("Tile isn't created from " + tile_dir + " with id " +
```

Comment:
im not really sure this is needed but a more appropriate message would be "could not load tile from disk"

Reply:
Done.